content/shmem_ctx.tex

```diff
@@ -0,0 +1,133 @@
+
+\begin{apiarguments}
+    \apiargument{IN}{options}{The set of options requested for the given context.
+        Multiple options may be requested by combining them with a bitwise
+        \textit{OR} operation.  Otherwise, $0$ can be given if no options are
+        requested.}
+    \apiargument{OUT}{ctx}{A handle to the newly created context.}
+\end{apiarguments}
+
+\apidescription{
+    The \FUNC{shmem\_ctx\_create} routine creates a new communication context
+    and returns its handle through the \VAR{ctx} argument.  If the context was
+    created successfully, a value of zero is returned; otherwise a nonzero
+    value is returned.  An unsuccessful context
+    creation call is not treated as an error and the \openshmem library remains
+    in a correct state.  The creation call can be reattempted with different
+    options or after additional resources become available.
+
+    By default contexts are {\em shareable} and, when it is allowed by the
+    threading model set during initialization, they can be used concurrently by
+    multiple threads within the PE where they were created.
+    %
+    The following options can be supplied during context creation to restrict
+    this usage model and enable performance optimizations.  When using a given
+    context, the application must comply with the requirements of all options
+    set on that context.  Otherwise the behavior is undefined.
+
+        \apitablerow{\CONST{SHMEM\_CTX\_SERIALIZED}}{
+            The given context will not be used by multiple threads
+            concurrently.  When the \CONST{SHMEM\_CTX\_SERIALIZED} option is
+            set, the user must ensure that operations involving the given
+            context are serialized by the application, including operations
+            that perform an explicit or implicit barrier.}
+
+        \apitablerow{\CONST{SHMEM\_CTX\_PRIVATE}}{
+            The given context will be used only by the thread that created it.
+            Contexts with the \CONST{SHMEM\_CTX\_PRIVATE} option set are
+            excluded from the quiet operation performed during explicit and
+            implicit barrier operations.}
+}
+
+\apireturnvalues{
+    Zero on success and nonzero otherwise.
+}
+
+\apinotes{
+    None.
+}
+
+\end{apidefinition}
+
+\subsubsection{\textbf{SHMEM\_CTX\_DESTROY}}
+\label{subsec:shmem_ctx_create}
+\apisummary{
+    Destroy a communication context.
+}
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+void shmem_ctx_destroy(shmem_ctx_t ctx);
+\end{Csynopsis}
+
+\begin{apiarguments}
+    \apiargument{IN}{ctx}{Handle to the context that will be destroyed.}
+\end{apiarguments}
+
+\apidescription{
+    \FUNC{shmem\_ctx\_destroy} destroys a context that was created by a call to
+    \FUNC{shmem\_ctx\_create}.  It is the user's responsibility to ensure that
+    the context is no longer in use before calling this routine.  This function
+    performs an implicit quiet operation on the given context before it is freed.
```

Comment:
I suppose it was added to handle the scenario where multiple threads use the same context handle. Users have to make sure such context handles are not being used by other threads, when one of the threads call destroy.